### ISSUE TITLE
Instructions to install PowerShell on OpenSuse 42.3 are not working for me.

### ISSUE THREAD

**Issue by TravisEz13** [MicrosoftDocs/PowerShell-Docs#2994]
Installing the package fails with `Problem: nothing provides openssl-libs needed by powershell-6.1.0-1.rhel.7.x86_64`

**Comment by kiazhi** [comment]
I tried to install PowerShell 6.1 on my openSUSE on Windows Subsystem for Linux (WSL) and do run into the same issue where `zypper` could not find the `openssl-libs` package.

```sh
> cat /etc/os-release
NAME="openSUSE Leap"
VERSION="42.3"
ID=opensuse
ID_LIKE="suse"
VERSION_ID="42.3"
PRETTY_NAME="openSUSE Leap 42.3"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:opensuse:leap:42.3"
BUG_REPORT_URL="https://bugs.opensuse.org"
HOME_URL="https://www.opensuse.org/"
```

I had to choose solution 2 to ignore the `openssl-libs` package dependency to continue the installation.

```sh
> sudo zypper install powershell
Loading repository data...
Reading installed packages...
Resolving package dependencies...

Problem: nothing provides openssl-libs needed by powershell-6.1.0-1.rhel.7.x86_64
 Solution 1: do not install powershell-6.1.0-1.rhel.7.x86_64
 Solution 2: break powershell-6.1.0-1.rhel.7.x86_64 by ignoring some of its dependencies

Choose from above solutions by number or cancel [1/2/c] (c): 2
Resolving dependencies...
Resolving package dependencies...

The following 2 NEW packages are going to be installed:
  libicu52_1 powershell

2 new packages to install.
Overall download size: 56.5 MiB. Already cached: 0 B. After the operation, additional 59.4 MiB will be used.
Continue? [y/n/...? shows all options] (y): y
Retrieving package libicu52_1-52.1-18.1.x86_64                                                                                                                                                          (1/2),   1.4 MiB (  4.2 MiB unpacked)
Retrieving: libicu52_1-52.1-18.1.x86_64.rpm .............................................................................................................................................................................[done (129.7 KiB/s)]
Retrieving package powershell-6.1.0-1.rhel.7.x86_64                                                                                                                                                     (2/2),  55.1 MiB ( 55.1 MiB unpacked)
Retrieving: powershell-6.1.0-1.rhel.7.x86_64.rpm ..........................................................................................................................................................................[done (1.2 MiB/s)]
Checking for file conflicts: ..........................................................................................................................................................................................................[done]
(1/2) Installing: libicu52_1-52.1-18.1.x86_64 .........................................................................................................................................................................................[done]
(2/2) Installing: powershell-6.1.0-1.rhel.7.x86_64 ....................................................................................................................................................................................[done]

> pwsh
PowerShell 6.1.0
Copyright (c) Microsoft Corporation. All rights reserved.

https://aka.ms/pscore6-docs
Type 'help' to get help.
```

After the installation, I ran the command to test if SSL is working.

```powershell
PS /home/ryentang> (Invoke-WebRequest -Uri "https://raw.githubusercontent.com/PowerShell/PowerShell/master/tools/metadata.json").Content
{
  "StableReleaseTag": "v6.1.0",
  "PreviewReleaseTag": "v6.1.0-rc.1",
  "ServicingReleaseTag": "v6.0.4",
  "ReleaseTag": "v6.1.0",
  "NextReleaseTag": "v6.1.0-rc.1"
}
```

Performed some further investigation and found that PowerShell 6.1 RPM package has a dependency for `openssl-libs` which is only available in RHEL, CentOS and Fedora distro.

```sh
$ zypper info --requires powershell
Loading repository data...
Reading installed packages...


Information for package powershell:
-----------------------------------
Repository     : msft_rhel_7
Name           : powershell
Version        : 6.1.0-1.rhel.7
Arch           : x86_64
Vendor         : Microsoft Corporation
Installed Size : 55.1 MiB
Installed      : No
Status         : not installed
Source package : powershell-6.1.0-1.rhel.7.src
Summary        : PowerShell is an automation and configuration management platform.
Description    :
    PowerShell is an automation and configuration management platform.
    It consists of a cross-platform command-line shell and associated scripting language.
Requires       : [3]
    /bin/sh
    libicu
    openssl-libs
```

Therefore, I think the problem is that there is no `openssl-libs` package from [openSUSE](https://software.opensuse.org/search?q=openssl-libs) repository.

By default, openSUSE will automatically install `libcurl4` package with its dependencies which will include `libssl.so.1.0.0`.

```sh
$ zypper info --requires libcurl4
Loading repository data...
Reading installed packages...


Information for package libcurl4:
---------------------------------
Repository     : oss_42.3_update
Name           : libcurl4
Version        : 7.37.0-39.1
Arch           : x86_64
Vendor         : openSUSE
Installed Size : 445.4 KiB
Installed      : Yes (automatically)
Status         : up-to-date
Source package : curl-7.37.0-39.1.src
Summary        : Version 4 of cURL shared library
Description    :
    The cURL shared library version 4 for accessing data using different
    network protocols.
Requires       : [14]
    libpthread.so.0()(64bit)
    libpthread.so.0(GLIBC_2.2.5)(64bit)
    libz.so.1()(64bit)
    libc.so.6(GLIBC_2.17)(64bit)
    libcrypto.so.1.0.0()(64bit)
    libssl.so.1.0.0()(64bit)
    libldap-2.4.so.2()(64bit)
    libgssapi_krb5.so.2()(64bit)
    libgssapi_krb5.so.2(gssapi_krb5_2_MIT)(64bit)
    liblber-2.4.so.2()(64bit)
    libssh2.so.1()(64bit)
    libidn.so.11()(64bit)
    libidn.so.11(LIBIDN_1.0)(64bit)
    /sbin/ldconfig
```

Where the `libssl.so.1.0.0` came from the `libopenssl1_0_0` package from [openSUSE](https://software.opensuse.org/search?q=libopenssl1_0_0) repository.

```sh
$ zypper info --provides --requires libopenssl1_0_0
Loading repository data...
Reading installed packages...


Information for package libopenssl1_0_0:
----------------------------------------
Repository     : oss_42.3_update
Name           : libopenssl1_0_0
Version        : 1.0.2j-29.1
Arch           : x86_64
Vendor         : openSUSE
Installed Size : 2.9 MiB
Installed      : Yes (automatically)
Status         : up-to-date
Source package : openssl-1.0.2j-29.1.src
Summary        : Secure Sockets and Transport Layer Security
Description    :
    The OpenSSL Project is a collaborative effort to develop a robust,
    commercial-grade, full-featured, and open source toolkit implementing
    the Secure Sockets Layer (SSL v2/v3) and Transport Layer Security (TLS
    v1) protocols with full-strength cryptography. The project is managed
    by a worldwide community of volunteers that use the Internet to
    communicate, plan, and develop the OpenSSL toolkit and its related
    documentation.

    Derivation and License

    OpenSSL is based on the excellent SSLeay library developed by Eric A.
    Young and Tim J. Hudson. The OpenSSL toolkit is licensed under an
    Apache-style license, which basically means that you are free to get it
    and to use it for commercial and noncommercial purposes.
Provides       : [6]
    libcrypto.so.1.0.0()(64bit)
    libssl.so.1.0.0()(64bit)
    libgost.so()(64bit)
    libpadlock.so()(64bit)
    libopenssl1_0_0 = 1.0.2j-29.1
    libopenssl1_0_0(x86-64) = 1.0.2j-29.1
Requires       : [5]
    libdl.so.2()(64bit)
    libz.so.1()(64bit)
    libdl.so.2(GLIBC_2.2.5)(64bit)
    libc.so.6(GLIBC_2.17)(64bit)
    /sbin/ldconfig
```

I think there might be a need to have a separate PowerShell 6.1 RPM package for openSUSE and SLES to correct the dependency?

**Comment by TravisEz13** [comment]
Or we could say just use the tar.gz or snap if you really want a package.

**Comment by kiazhi** [comment]
I think it will preferable to have an `tar.gz` installation steps for openSUSE instead of snap because not all enterprise environment like additional package manager.

Do you want me to help updating the doc for openSUSE?

**Comment by TravisEz13** [comment]
That would be be very helpful.  I can point you to our docker image which uses the tar.gz (because I could not get the package to work.)
https://github.com/PowerShell/PowerShell-Docker/blob/master/release/stable/opensuse423/docker/Dockerfile

**Comment by kiazhi** [comment]
@TravisEz13, I used that [Dockerfile](https://github.com/PowerShell/PowerShell-Docker/blob/master/release/stable/opensuse423/docker/Dockerfile) that you provided to build the container image and tested the `pwsh` interactively in the container without issue locally on two machine (Windows10 & macOS).

I went about looking at the Azure DevOps Pipelines for the failed openSUSE job and find out where the issue is.

From the log, it seems like your openSUSE build has an issue with `zypper update` at that time. I don't know why you got a permission to access denied to that file.

```text
Permission to access 'http://download.opensuse.org/distribution/leap/42.3/repo/non-oss/suse/repodata/repomd.xml' denied.

The command '/bin/sh -c zypper --non-interactive update --skip-interactive' returned a non-zero code: 106
```

So I don't think that issue is with `tar.gz` or the dependency packages. But more of the `zypper update`?

**Comment by TravisEz13** [comment]
I've seen this issue intermittently.  I don't know what to do about it.

**Comment by kiazhi** [comment]
Unfortunately, it seems to be intermittent issue from the openSUSE repository end.

**Comment by sdwheeler** [comment]
Fixed in PR#3163
